### PR TITLE
Show support for current Nova version

### LINKED ISSUE #196: Show support for Nova version 3 or 4
Now that Nova 4 is out in the wild, I believe it is useful to add some metadata to each packages to mark its support for version 3 or 4
Also it would be nice if the packages could be filtered on their version support

### PR DESCRIPTION
**Pull request by RhysLees**
# This PR adds the following:
-  nova_version to packages DB Table
- Nova version to PackageDetailFrame if their latest stable release requires a Nova version
image
- Current Nova Version Supported Label to Package cards if their latest stable release requires a Nova version that matches the major version specified in env 
image

- Adds Nova's current filter to tags and filters their latest stable release if it requires a Nova version that matches the major version specified in env 
image

When a new major version of Nova is released all that is needed is to increase the version in .env from `NOVA_LATEST_MAJOR_VERSION=4` to `NOVA_LATEST_MAJOR_VERSION=5`

Closes #196

### CODE REVIEW

File: app/Http/Resources/PackageDetailResource.php
Comment (dmason30) [inline app/Http/Resources/PackageDetailResource.php:27]: Also check `require-dev`? 

Code could also be simplified to null coalesce, this is personal preference though:
```php
$novaVersion = $composer_latest['require']['laravel/nova']
    ?? $composer_latest['require-dev']['laravel/nova']
    ?? null
```

Comment (RhysLees) [reply app/Http/Resources/PackageDetailResource.php:27]: Agreed I'll make the changes shortly

Comment (RhysLees) [reply app/Http/Resources/PackageDetailResource.php:27]: > Agreed I'll make the changes shortly.
> 
> In regards to the require-dev I believe we came to the agreement not to show nova 4 support unless it was specifically in require.

Comment (dmason30) [reply app/Http/Resources/PackageDetailResource.php:27]: Its up to you guys, but I have seen some release 4.x support by only including it in require-dev, not sure of the reasoning myself.

Comment (RhysLees) [reply app/Http/Resources/PackageDetailResource.php:27]: I think it will be best if we stick to only `require` for now and if it becomes an issue where `require-dev` is needed then we could add that in

**Comment by RhysLees** [comment]
@marcusmoore Is there anything else you think should be added?

File: config/novapackages.php
Comment (marcusmoore) [inline config/novapackages.php:14]: Can we move this above the `latest_major_version` a couple lines below?

File: app/Jobs/SyncPackagePackagistData.php
Comment (marcusmoore) [inline app/Jobs/SyncPackagePackagistData.php:46]: Can we remove the extra line here?

File: app/Jobs/SyncPackagePackagistData.php
Comment (marcusmoore) [inline app/Jobs/SyncPackagePackagistData.php:72]: I don't think we need ` ?? null` here right?

```suggestion
                'nova_version' => $novaVersion,
```

**Review by marcusmoore — CHANGES_REQUESTED** [review]
This looks good! I have some minor questions/changes but overall 👍🏾 

Thank you 😄

File: app/Jobs/SyncPackagePackagistData.php
Comment (RhysLees) [reply app/Jobs/SyncPackagePackagistData.php:72]: Sure, will make the changes tonight 😃

**Comment by RhysLees** [comment]
@marcusmoore I can also close #63 with the same type of tag instead of saying `Potentially Abandoned` Would it be worth adding to this PR or as a seperate PR?

**Comment by marcusmoore** [comment]
@RhysLees thanks for the updates and also peeking at additional issues that might be solved in a similar manner. A separate PR would be appreciated.

I'll review (and hopefully merge) this PR later this week.

Thanks again! 😄

**Comment by RhysLees** [comment]
> @RhysLees thanks for the updates and also peeking at additional issues that might be solved in a similar manner. A separate PR would be appreciated.
> 
> I'll review (and hopefully merge) this PR later this week.
> 
> Thanks again! 😄 

Awesome, thanks fore Reviewing, I'll wait until this is merged then I'll make a PR for the other issue as it mainly uses the same code from this PR 😊

**Comment by RhysLees** [comment]
> > @RhysLees thanks for the updates and also peeking at additional issues that might be solved in a similar manner. A separate PR would be appreciated.
> > 
> > I'll review (and hopefully merge) this PR later this week.
> > 
> > Thanks again! 😄 
> 
> Awesome, thanks for Reviewing, I'll wait until this is merged then I'll make a PR for the other issue as it mainly uses the same code from this PR 😊

**Review by marcusmoore — APPROVED** [review]

**Comment by marcusmoore** [comment]
@RhysLees hey look at that!

CleanShot 2022-05-06 at 10 49 29

This has been deployed. Thank you for your work on this 😄

**Comment by RhysLees** [comment]
> @RhysLees hey look at that!
> 
> CleanShot 2022-05-06 at 10 49 29
> 
> This has been deployed. Thank you for your work on this 😄

AYYYYYYY let's go! Thanks for your help in making it possible!

**Pull request merged**
